### PR TITLE
Register user activities in background

### LINKED ISSUE #51: Register user activities in background
Initial comment - https://github.com/fs/rails-base-graphql-api/pull/49#discussion_r438079589

Create activities (for example - user registered) in background. The main process shouldn't fail if activity couldn't be created.

### PR DESCRIPTION
**Pull request by AzatZaynutdinov**
### Summary

Register user activities in background

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

* Open GraphiQL App
* Make request using schema:
```
  query {
    me: {
      id
      name
    }
  }
```
* ...

### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, ENV variables, services, scripts, etc.

### References
* Resolves #51 Issue

### CODE REVIEW

File: app/jobs/register_activity_job.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class RegisterActivityJob < ApplicationJob
+  queue_as :events
+
+  def perform(user_id)
+    user = User.find(user_id)
```
Comment (EvgeniyEsaulkov) [inline app/jobs/register_activity_job.rb:5]: What if the record isn't found? For example it was deleted while this job was in queue. Should we use `rescue_from` for this case or it is OK to leave job as failed?

File: spec/jobs/register_activity_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+describe RegisterActivityJob do
```
Comment (EvgeniyEsaulkov) [inline spec/jobs/register_activity_job_spec.rb:3]: What do you think if we'll test that this job calls `CreateRegisterActivity` interactor? Something like this:
```ruby
it "calls interactor to create activity" do
  expect(CreateRegisterActivity).to receive(:call).with(user: user)

  described_class.perform_now(user)
end
```

Comment (ArthurZaharov) [reply spec/jobs/register_activity_job_spec.rb:3]: I think that this test should cover two scenarious
```
context "when user exists" do
  it "calls interactor to create activity" do
    ...
  end
end

context "when user does not exist" do
  it "raises" do
    ...
  end
end```

File: spec/jobs/register_activity_job_spec.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  include ActiveJob::TestHelper
+
+  let!(:user) { create :user }
+
+  before do
+    ActiveJob::Base.queue_adapter = :test
+  end
+
+  it "matches with enqueued job" do
```
Comment (ArthurZaharov) [inline spec/jobs/register_activity_job_spec.rb:12]: Would it be reasonable to move this spec to the sign up user organizer spec where we enqueue the job?

File: spec/jobs/register_activity_job_spec.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+require "rails_helper"
+
+describe RegisterActivityJob do
+  include ActiveJob::TestHelper
+
+  let!(:user) { create :user }
+
+  before do
+    ActiveJob::Base.queue_adapter = :test
```
Comment (ArthurZaharov) [inline spec/jobs/register_activity_job_spec.rb:9]: Should we move the configuration in the `rails_helper.rb`?

File: app/interactors/create_user.rb [hunk 5]
```diff
@@ -9,6 +9,10 @@ def call
     context.fail!(error_data: error_data) unless context.user.save
   end
 
+  after do
```
Comment (EvgeniyEsaulkov) [inline app/interactors/create_user.rb:12]: Could you please move this webhook to `SignUp` organizer? This way the interactor will be responsible for user creation only. And organizer calls all actions required for sign up: create user,  create access/refresh tokens, call job to register activity.

File: spec/jobs/register_activity_job_spec.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+      described_class.perform_now(user.id)
+    end
+  end
+
+  context "when user does not exist" do
+    it "interactor will not create activity" do
+      expect(CreateRegisterActivity).not_to receive(:call)
+
+      described_class.perform_now(777)
```
Comment (EvgeniyEsaulkov) [inline spec/jobs/register_activity_job_spec.rb:20]: Could you please make the argument more descriptive? For example:
```ruby
let(:wrong_user_id) { 777 }
...
described_class.perform_now(wrong_user_id)
```

**Pull request merged**